boa/src/builtins/function/mod.rs

```diff
@@ -37,15 +37,21 @@ pub struct RegularFunction {
     /// The fields associated with the function
     pub object: Object,
     /// This function's expression
-    pub expr: Expr,
+    pub expr: Node,
```

Comment:
I would change the name of this to `node`, if it's going to contain a node.

Reply:
> This looks very nice! Do you want me to add the changes from #263?

Hold off for now, when i get further in to the point a majority of this works then yep, right now its still a bit of a mess